src/transaction/submit.js

```diff
@@ -27,7 +27,10 @@ function formatResponse(response) {
   if (isImmediateRejection(response.engine_result)) {
     throw new utils.common.errors.RippledError('Submit failed');
   }
-  return convertKeysFromSnakeCaseToCamelCase(response);
+  return {
+    resultCode: response.engine_result,
+    resultMessage: response.engine_result_message
+  };
```

Comment:
The transaction hash is returned by `sign`. What is it convenient for? Shouldn't you already know what you submitted? None of the other methods redundantly return the request data.